### PR TITLE
Replace boto with boto3 (closes #73)

### LINKED ISSUE #73: Adopt boto3
Even though tests are passing, boto2 technically only supports up to Python 3.4. boto3 supports Python 3.7+.

┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DAL-51)
┆Project Name: dos-azul-lambda
┆Issue Number: DAL-51
┆Epic: DAL One point ohh

### PR DESCRIPTION
**Pull request by natanlao**
This commit updates app.py and specified dependencies to use
boto3 over boto, which allows us to improve our Python 3.x support.
In the process of doing so, "manual" calls to ElasticSearch have
been replaced with the ElasticSearch Python bindings, which provide
more robust logging and error handling.

The provisioning script is also updated to use boto3 and the
ElasticSearch bindings.

### CODE REVIEW

**Comment by natanlao** [comment]
This also fixes an (unreported) bug where the `is_secure=False` flag was used when starting the ElasticSearch connection.

File: provision/provision.py [hunk 1]
```diff
@@ -56,27 +50,19 @@ def get_endpoint(es_domain):
...
+    # Set up indexes
     with open(getpath('index-mapping.json'), 'r') as data:
-        payload = json.load(data)['fb_index']
-    make_es_request(method='PUT', path='/' + obj_index, data=json.dumps(payload))
-
-    # Set up db_index (data bundles)
-    with open(getpath('index-mapping.json'), 'r') as data:
-        payload = json.load(data)['db_index']
-    make_es_request(method='PUT', path='/' + bdl_index, data=json.dumps(payload))
+        index_cfg = json.load(data)
```
Comment (david4096) [inline provision/provision.py]: This is much nicer.

**Comment by david4096** [comment]
Great work @natanlao ! This keeps this repository free of legacy code and aligns better with the approach azul has to working with elasticsearch. Using the boto2 client to work with elasticsearch was awkward. Great to see PRs that remove code and make improvements!

**Comment by natanlao** [comment]
Thanks for the review!

**Pull request merged**